pkg/scalers/gcp_common.go

```diff
@@ -0,0 +1,42 @@
+package scalers
```

Comment:
Minor nit: could you please move this file to a separte package `gcp`? We have the same for common `azure` stuff.

Reply:
This will create a dependency cycle as the getGcpAuthorization method uses ScalerConfig

Reply:
:(  right

it's probably a time to move ScalerConfig and relevant stuff to standalone package